torch/_subclasses/meta_utils.py

```diff
@@ -327,16 +329,39 @@ def empty_create(inner_t, inner_src, symbolic_context=symbolic_context):
                     if t.requires_grad and not is_leaf:
                         with torch.enable_grad():
                             r = r.clone()
-                elif is_batchedtensor(t):
-                    # Wraps a BatchedTensor in a FakeTensor
+                elif is_functorch_wrapped_tensor(t):
+                    # Wraps a functorch tensor class (BatchedTensor, GradTrackingTensor)
+                    # in a FakeTensor
                     def _to_fake_tensor(t):
                         if is_batchedtensor(t):
                             ft = _to_fake_tensor(get_unwrapped(t))
                             lvl = maybe_get_level(t)
                             bdim = maybe_get_bdim(t)
                             r = _add_batch_dim(ft, bdim, lvl)
+                        elif is_gradtrackingtensor(t):
+                            disable_functorch = torch._C._DisableFuncTorch
+                            with disable_functorch():
+                                ft = _to_fake_tensor(get_unwrapped(t))
+                            lvl = torch._C._functorch.maybe_get_level(t)
+                            r = torch._C._functorch._wrap_for_grad(ft, lvl)
+
+                            if t._is_view():
+                                # is this correct?
+                                # it fails if r is a nested functorch tensor:
+                                # r = GradTrackingTensor(
+                                #       BatchedTensor(
+                                #           Tensor
+                                #       )
+                                #   )
+                                r = r.view_as(r)
```

Comment:
From my understanding, FakeTensor has a mechanism for modeling view relationships. Could we figure out how it does it for regular tensors and figure out how to apply it to GradTrackingTensor?

Reply:
Hmm, I have a better question: when do we actually run this fakeification logic for GradTrackingTensor?

I would expect that FakeTensor propagation (e.g. calling GradTrackingTensor(FakeTensor).sin() just works). So it's unclear to me how we get to this point